### PR TITLE
[Java] Feat add java builder pattern

### LINKED ISSUE #18032: [REQ] [java] create a builder pattern for the models
### Is your feature request related to a problem? Please describe.

A good builder pattern generation can solve many issues:
* initialize readonly values in the java generator
* make it similar to the lombok builder that people are used to.
* introduce immutability

Lombok can be used the generate builders.  But the introduction of lombok creates more issues (#17793)

## Describe the solution you'd like

Add a builder inner class
Add a static builder() method
Add a toBuilder() method

## Describe alternatives you've considered

Reuse the builder from javaJAXRSServerCodeGen.
Fix the lombok generation issues.

### PR DESCRIPTION
**Pull request by jpfinne**
Fix #18032.  Add a new configOptions:  generateBuilders to java client and spring generators 

The name of the option is the same as in JavaJAXSRSSpecServerCodeGen.

Spring + lombok:  generateBuilders + generateConstructorWithAllArgs configOptions can be used instead of lombok.

The builder is not created when lombok.Data is configured (issue with fluent setters not generated)

Improve generation of no args constructor.
Also minor fix for the generateConstructorWithAllArgs and Lombok.AllArgsConstructor (there was some typos)

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the 7.0.6 (https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Please review
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @martin-mfg (2023/08)
Java Spring | @cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09) @martin-mfg (2023/08)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache [hunk 1]
```diff
@@ -151,7 +151,7 @@ public class {{classname}}{{#parent}} extends {{{parent}}}{{/parent}}{{^parent}}
   {{! begin feature: fluent setter methods }}
   public {{classname}} {{name}}({{{datatypeWithEnum}}} {{name}}) {
     {{#openApiNullable}}
-    this.{{name}} = {{#isNullable}}JsonNullable.of({{/isNullable}}{{#useOptional}}{{^required}}{{^isNullable}}{{^isContainer}}Optional.of({{/isContainer}}{{/isNullable}}{{/required}}{{/useOptional}}{{name}}{{#isNullable}}){{/isNullable}}{{#useOptional}}{{^required}}{{^isNullable}}{{^isContainer}}){{/isContainer}}{{/isNullable}}{{/required}}{{/useOptional}};
+    this.{{name}} = {{#isNullable}}JsonNullable.of({{/isNullable}}{{#useOptional}}{{^required}}{{^isNullable}}{{^isContainer}}Optional.ofNullable({{/isContainer}}{{/isNullable}}{{/required}}{{/useOptional}}{{name}}{{#isNullable}}){{/isNullable}}{{#useOptional}}{{^required}}{{^isNullable}}{{^isContainer}}){{/isContainer}}{{/isNullable}}{{/required}}{{/useOptional}};
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache:154]: is this change required for the builder pattern?

Comment (jpfinne) [reply modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache]: I've reverted the change.

I will create a new issue for this useOptional flag described as " Use Optional container for optional parameters" .
It creates optional fields that can lead to NullPointerException...
It also makes the mustache template so harder to maintain

**Comment by wing328** [comment]
do you mind filing a new one instead (using the latest master of the official openapi-generator) as some commits do not seem to be related to this change at all?

**Comment by jpfinne** [comment]
> do you mind filing a new one instead (using the latest master of the official openapi-generator) as some commits do not seem to be related to this change at all?

@wing328 See #18648
